gnmlibrarytool/urls.py

```diff
@@ -30,4 +31,12 @@
     url(r'^diagram$', login_required(DiagramMainView.as_view())),
     url(r'^endpoint/', include(router.urls)),
     url(r'^storageruleinfo/(\w{2}-\d+)$',StorageRuleInfoView.as_view(), name="libtool_storage_info_view"),
+    url(r'^rules/$', login_required(rule_form), name='rules'),
+    url(r'^add/$', login_required(add_rule)),
+    url(r'^addrule/$', login_required(add_rule_to_item)),
+    url(r'^delete/$', login_required(delete_rule)),
```

Comment:
I am not sure what you mean by override views. 

I will take out one of the lines in urls.py which is pointless.

What does 'defining views in the models' mean?

'add' is for adding a rule to the model that holds rules for later use in the Portal interface.

'addrule' if for adding one of these rules to a Vidispine item.